lib/modules/dosomething/dosomething_campaign/includes/Campaign.php

```diff
@@ -0,0 +1,285 @@
+    if ($data['sizes']['landscape']) {
+      $data['sizes']['landscape']['uri'] = dosomething_image_get_themed_image_url($image->nid, 'landscape', '1440x810');
+    }
+
+    if ($data['sizes']['square']) {
+      $data['sizes']['square']['uri'] = dosomething_image_get_themed_image_url($image->nid, 'square', '300x300');
+    }
+
+    return $data;
+  }
+
+
+  protected function getCoverImageAlt() {
+    $image_id = $this->variables['alt_image_campaign_cover_nid'];
+
+    if ($image_id) {
+      return $this->getCoverImage($image_id);
+    }
+    else {
+      return NULL;
+    }
+  }
+
+
+  protected function getFactData() {
+    $data = array();
+
+    $fact_fields = array('field_fact_problem', 'field_fact_solution');
+    $fact_vars = dosomething_fact_get_mutiple_fact_field_vars($this->node, $fact_fields);
+
+    if (!empty($fact_vars)) {
+      foreach ($fact_vars['facts'] as $index => $fact_data) {
+        $index = str_replace('field_', '', $index);
+
+        $data[$index]['fact'] = dosomething_helpers_issetor($fact_data['fact']);
+        $data[$index]['id'] = dosomething_helpers_issetor($fact_data['nid']);
+        $data[$index]['footnote'] = (int) dosomething_helpers_issetor($fact_data['footnotes']);
+        $data[$index]['source'] = dosomething_helpers_issetor($fact_data['sources']);
+
+      }
+
+      $data['sources'] = dosomething_helpers_issetor($fact_vars['sources']);
+
+      return $data;
+    }
+
+    return NULL;
+
+  }
+
+
+  protected function getIssue() {
+    // @TODO: Need to find out if this is allowed to be a field with multiple values...?
+    $issue_id = dosomething_helpers_extract_field_data($this->node->field_issue);
+
+    if ($issue_id) {
+      $issue = $this->getTaxonomyTerm($issue_id);
+
+      return $issue;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getPrimaryCause() {
+    $cause_id = dosomething_helpers_extract_field_data($this->node->field_primary_cause);
+
+    if ($cause_id) {
+      $cause = $this->getTaxonomyTerm($cause_id);
+
+      return $cause;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getTags() {
+    $data = array();
+
+    $tag_ids = dosomething_helpers_extract_field_data($this->node->field_tags);
+
+    if ($tag_ids) {
+      foreach ($tag_ids as $tid) {
+        $data[] = $this->getTaxonomyTerm($tid);
+      }
+
+      return $data;
+    }
+
+    return NULL;
+  }
+
+
+  protected function getTaxonomyTerm($tid) {
+    $data = array();
+
+    $taxonomy = taxonomy_term_load($tid);
+
+    $data['id'] = $taxonomy->tid;
+    $data['name'] = strtolower($taxonomy->name);
+
+    return $data;
+  }
+
+
+  protected function getTagline() {
+    return dosomething_helpers_extract_field_data($this->node->field_call_to_action);
+  }
+
+
+  /**
+   * @return string
+   */
+  protected function getTimeCommitment() {
+    // @TODO: I've renamed "active_hours" to "time_commitment" because it sounds more straightforward; but appreciate feedback.
```

Comment:
We've chatted about it, but to clarify with others, `call_to_action` is changing to `tagline` because we're already doing that on the front-end and design side. A "call to action" is not actually data, but a design pattern that applies to a few things on the front-end, so to remove confusion there was a renaming. Now just trying to bring that into the backend as well.